### PR TITLE
Fix lints

### PR DESCRIPTION
**Pull request by MikailBag**

### CODE REVIEW

**Comment by MikailBag** [comment]
PR should become green with the next nightly toolchain

File: kube-runtime/src/controller/mod.rs [hunk 1]
```diff
@@ -37,7 +37,7 @@ mod future_hash_map;
 mod runner;
 
 #[derive(Debug, Error)]
-pub enum Error<ReconcilerErr: std::error::Error + 'static, QueueErr: std::error::Error + 'static> {
+pub enum Error<ReconcilerErr: 'static, QueueErr: 'static> {
```
Comment (clux) [inline kube-runtime/src/controller/mod.rs:40]: oh, huh, does this actually work :thinking:

Comment (MikailBag) [reply kube-runtime/src/controller/mod.rs:40]: Thiserror automatically adds `...: Error` bounds when generating `impl Error`, so these bounds were not necessary.

**Comment by MikailBag** [comment]
Hmm, I guess I misunderstood how clippy gets released. Looks like it's still buggy.
I've added explicit `allow` for the false positive.

**Comment by teozkr** [comment]
Thanks!

**Pull request merged**